### PR TITLE
Yet another MCMC processor update

### PR DESCRIPTION
**Pull request by KSkwarczynski**
moving stuff from T2K

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,9 +47,9 @@ CPMFindPackage(
 # In the future which osc calc we use might be set with a flag
 CPMFindPackage(
     NAME Prob3plusplus
-    VERSION 3.10.3
```
Comment (EdAtkin) [inline CMakeLists.txt:50]: do you know what changes to 3.10.4?

File: mcmc/MCMCProcessor.h [hunk 2]
```diff
@@ -77,21 +80,21 @@ class MCMCProcessor {
     void ResetHistograms();
     
     // Get the number of parameters
-    int GetNParams() { return nDraw; };
-    int GetNFlux() { return nFlux; };
-    int GetNXSec() { return nParam[kXSecPar]; };
-    int GetNND() { return nParam[kND280Par]; };
-    int GetNFD() { return nParam[kFDDetPar]; };
-    int GetOSC() { return nParam[kOSCPar]; };
+    inline int GetNParams() { return nDraw; };
```
Comment (EdAtkin) [inline mcmc/MCMCProcessor.h:83]: I'm guessing these are all just inline for a small speed increase?

Comment (KSkwarczynski) [reply mcmc/MCMCProcessor.h:83]: Exactly

File: splines/gpuSplineUtils.cu [hunk 3]
```diff
@@ -19,7 +19,10 @@
 
 // Hard code the number of splines
 // Not entirely necessary: only used for val_gpu and segment_gpu being device constants. Could move them to not being device constants
-#define __N_SPLINES__ 48
+// EM: for OA2022:
+// #define __N_SPLINES__ 48
+// EM: for OA2024:
+#define __N_SPLINES__ 200
```
[the remark this answers is not in the document]
Comment (EdAtkin) [reply splines/gpuSplineUtils.cu:25]: ok cool. This is fine :) just the user will have to be aware of this but not one to fix right now.

**Pull request merged**